src/goto-symex/symex_goto.cpp

```diff
@@ -48,8 +48,8 @@ void goto_symext::symex_goto(statet &state)
     !instruction.targets.empty(), "goto should have at least one target");
 
   // we only do deterministic gotos for now
-  if(instruction.targets.size()!=1)
-    throw "no support for non-deterministic gotos";
+  DATA_INVARIANT(
+  	instruction.targets.size()!=1, "no support for non-deterministic gotos");
```

Comment:
Not entirely sure if this is a data invariant - the error message seems to suggest that it's valid to construct such a thing, it's just the solver that doesn't have support for it.

Reply:
I think this should stay, it's just that this must be inverted.